wire-runtime/src/jvmMain/kotlin/com/squareup/wire/Messages.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+import kotlin.reflect.KProperty1
+import okio.Buffer
+import okio.IOException
+
+/**
+ * Reads through the [message's unknown fields][Message.unknownFields] and return the first value
+ * matching the [property]'s tag as set in the [WireField] annotation attached to it. If the tag is
+ * not found, it will return null. If the tag is found but the value attached to it isn't a varint,
+ * the method will throw an [IOException].
+ */
+@Throws(IOException::class)
+inline fun <reified M : Message<M, B>, B : Message.Builder<M, B>, reified P : Enum<P>> Message<M, B>.unknownEnumTag(property: KProperty1<M, P?>): Int? {
+  val target = M::class.java.getField(property.name).getAnnotation(WireField::class.java).tag
+
+  val reader = ProtoReader(Buffer().write(unknownFields))
+  reader.forEachTag { tag ->
+    if (tag != target) {
+      reader.skip()
+    } else {
+      return reader.readVarint32()
+    }
+  }
+
+  return null
+}
+
+/**
+ * Reads through the [message's unknown fields][Message.unknownFields] and return all values
+ * matching the [property]'s tag as set in the [WireField] annotation attached to it. If the tag is
+ * found but the value attached to it isn't a varint, the method will throw an [IOException].
+ */
+@Throws(IOException::class)
+inline fun <reified M : Message<M, B>, B : Message.Builder<M, B>, reified P : Enum<P>> Message<M, B>.unknownEnumTags(property: KProperty1<M, List<P>>): List<Int> {
```

Comment:
This is for repeated properties?

Reply:
Yes, `repeated MyEnum my_enums = 1`

Reply:
I wanna rename it to `unknownEnumValue` and `unknownEnumValues` by the way.